modules/3d/src/pointcloud/io_obj.cpp

```diff
@@ -75,9 +76,10 @@ void ObjDecoder::readData(std::vector<Point3f> &points, std::vector<Point3f> &no
     file.close();
 }
 
-void ObjEncoder::writeData(const std::vector<Point3f> &points, const std::vector<Point3f> &normals, const std::vector<std::vector<int32_t>> &indices)
+void ObjEncoder::writeData(const std::vector<Point3f> &points, const std::vector<Point3f> &normals, const std::vector<Point3_<uchar>> &rgb, const std::vector<std::vector<int32_t>> &indices)
 {
     std::ofstream file(m_filename, std::ios::binary);
+    CV_UNUSED(rgb);
```

Comment:
It's very easy to add color information to OBJ format too. I propose to do it in this PR. See https://en.wikipedia.org/wiki/Wavefront_.obj_file

Reply:
Hi asmorkalov, sorry for the late reply. I'm a bit confused because for .obj files, there doesn't seem to be a clear command for color information. The OBJ files are usually defined by polygon faces (which themselves are defined by vertexes, or points) and normals, curves, texture maps, and surfaces. So I'm wondering if it is still necessary to add color read/write for .obj files.

Reply:
Wkipedia is not 100% source of truth for the case, but it defines the following:

> A [vertex](https://en.wikipedia.org/wiki/Vertex_(geometry)) is specified via a line starting with the letter v. That is followed by (x,y,z[,w]) coordinates. W is optional and defaults to 1.0. A right-hand coordinate system is used to specify the coordinate locations. Some applications support vertex colors, by putting red, green and blue values after x y and z (this precludes specifying w). The color values range from 0 to 1.[[2]](https://en.wikipedia.org/wiki/Wavefront_.obj_file#cite_note-2)

I think it's not a problem to treat the last 3 values as colors, if pixel value contains 6 or 7 elements.
I propose it to have more-or-less similar support level for PLY and OBJ, if it's possible.